packages/react-component-library/src/components/DataGrid/DataGrid.tsx

```diff
@@ -78,9 +82,29 @@ export interface DataGridPropsWithInternalSorting<T extends object>
   sorting?: never
 }
 
+export interface DataGridPropsWithExternalPagination<T extends object>
+  extends DataGridBaseProps<T>,
+    Pick<TableOptions<T>, 'onPaginationChange'> {
+  manualPagination: true
```

Comment:
Do we need `manualPagination` or can we just know that the pagination is externally controlled if `onPaginationChange` is defined?

Reply:
I've changed it slightly so `onPaginationChange` gets invoked regardless of whether `manualPagination` is set.

This way even with internal pagination state there is still a handler provide feedback.

This is more in line with the original behaviour.